tests/rptest/tests/delete_records_test.py

```diff
@@ -775,3 +783,153 @@ class PartitionMoveExceptionReporter:
         status = consumer.consumer_status
         assert status.validator.invalid_reads == 0
         assert status.validator.out_of_scope_invalid_reads == 0
+
+
+class DeleteRecordsKafkaTest(RedpandaTest):
+    """
+    To assert DeleteRecords has the same semantics with a Kafka client.
+    Uses kafka-delete-records.sh and rpk trim-prefix directly to assert
+    we achieve parity with Kafka.
+    """
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context, num_brokers=3)
+        self.ctx = test_context
+        self.rpk = RpkTool(self.redpanda)
+        self.admin = Admin(self.redpanda)
+        self.client = DefaultClient(self.redpanda)
+
+    def get_topic_info(self, topic_name):
+        topics_info = list(self.rpk.describe_topic(topic_name))
+        self.logger.info(topics_info)
+        assert len(topics_info) == 1
+        return topics_info[0]
+
+    def make_delete_records_for_cli(self,
+                                    topic_partition_offsets: list[tuple[str,
+                                                                        int,
+                                                                        int]]):
+        delete_records_json = {"version": 1, "partitions": []}
+        for t, p, o in topic_partition_offsets:
+            delete_records_json["partitions"].append({
+                "topic": t,
+                "partition": p,
+                "offset": o
+            })
+        return delete_records_json
+
+    def execute_delete_records(self, delete_records_json: dict):
+        kafka_tools = KafkaCliTools(self.redpanda)
+        res = kafka_tools.delete_records(delete_records_json)
+        split_str = res.split('\n')
+        assert split_str[-1] == ""
+        partition_watermark_lines = split_str[2:-1]
+        result_list: list[RpkTrimOffsetResponse] = []
+        for partition_watermark_line in partition_watermark_lines:
+            topic_partition_str, result_str = partition_watermark_line.strip(
```

Comment:
extract the parsing into a function please otherwise this code is barely readable

Reply:
maybe this parsing needs to go into KafkaCliTools?

Reply:
Moved parsing back into `KafkaCliTools.delete_records()`.